commit d72767c900e448bb7cf14adf21629c868054de0f
Author: Markus Bauer <mb@karo-electronics.de>
Date:   Thu Sep 17 13:05:59 2020 +0000

    gpu: drm: imx: Use correct bus_flags from drm_display_info and also fix bug in function that checks them

diff --git a/drivers/gpu/drm/imx/parallel-display.c b/drivers/gpu/drm/imx/parallel-display.c
index 08fafa4bf8c2..ac3a2c407b97 100644
--- a/drivers/gpu/drm/imx/parallel-display.c
+++ b/drivers/gpu/drm/imx/parallel-display.c
@@ -226,9 +226,21 @@ static int imx_pd_bridge_atomic_check(struct drm_bridge *bridge,
 		return -EINVAL;
 
 	if (bus_flags &
-	    ~(DRM_BUS_FLAG_DE_LOW | DRM_BUS_FLAG_DE_HIGH |
+	    ~(DRM_BUS_FLAG_DE_LOW |
+	      DRM_BUS_FLAG_DE_HIGH |
+	      DRM_BUS_FLAG_PIXDATA_DRIVE_POSEDGE |
 	      DRM_BUS_FLAG_PIXDATA_DRIVE_POSEDGE |
-	      DRM_BUS_FLAG_PIXDATA_DRIVE_NEGEDGE)) {
+	      DRM_BUS_FLAG_PIXDATA_DRIVE_NEGEDGE |
+	      DRM_BUS_FLAG_PIXDATA_SAMPLE_POSEDGE |
+	      DRM_BUS_FLAG_PIXDATA_SAMPLE_NEGEDGE |
+	      DRM_BUS_FLAG_DATA_MSB_TO_LSB |
+	      DRM_BUS_FLAG_DATA_LSB_TO_MSB |
+	      DRM_BUS_FLAG_SYNC_DRIVE_POSEDGE |
+	      DRM_BUS_FLAG_SYNC_DRIVE_NEGEDGE |
+	      DRM_BUS_FLAG_SYNC_SAMPLE_POSEDGE |
+	      DRM_BUS_FLAG_SYNC_SAMPLE_NEGEDGE |
+	      DRM_BUS_FLAG_SHARP_SIGNALS)
+	) {
 		dev_warn(imxpd->dev, "invalid bus_flags (%x)\n", bus_flags);
 		return -EINVAL;
 	}
